src/main/java/org/opensearch/knn/index/engine/MethodComponentContext.java

```diff
@@ -49,6 +49,29 @@ public class MethodComponentContext implements ToXContentFragment, Writeable {
     private final String name;
     private final Map<String, Object> parameters;
 
+    /**
+     * Copy constructor. Creates a deep copy of a {@link MethodComponentContext}
+     *
+     * @param methodComponentContext to be copied. Must NOT be null
+     */
+    public MethodComponentContext(MethodComponentContext methodComponentContext) {
+        if (methodComponentContext == null) {
+            throw new IllegalArgumentException("MethodComponentContext cannot be null");
+        }
+
+        this.name = methodComponentContext.name;
+        this.parameters = new HashMap<>();
+        if (methodComponentContext.parameters != null) {
+            for (Map.Entry<String, Object> entry : methodComponentContext.parameters.entrySet()) {
+                if (entry.getValue() instanceof MethodComponentContext) {
+                    parameters.put(entry.getKey(), new MethodComponentContext((MethodComponentContext) entry.getValue()));
+                } else {
+                    parameters.put(entry.getKey(), entry.getValue());
```

Comment:
They'll be one of the following: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/engine/Parameter.java. So it will be okay